### PR TITLE
(integrations restructure) add notification & incidents folder

### PR DESCRIPTION
**Pull request by imatwawana**
This PR updates the navigation of the Integrations section (1/9), doing the following:

- Adds a Notification & Incidents folder under Integrations
- Moves pages for Amixr, Microsoft Teams, PagerDuty, Slack, and Spike.sh under new folder
- Updates sidebar order of all integrations pages to support new folder structure
- Adds redirects and updates links
- Adds Spike.sh to associated table on main Integrations page and makes table alphabetical

### CODE REVIEW

File: src/docs/product/alerts/create-alerts/issue-alert-config.mdx [hunk 1]
```diff
@@ -58,9 +58,9 @@ Learn more about and <PlatformLink to="/enriching-events/tags/">tags</PlatformLi
 
 - Send a notification to either [Issue Owners](/product/error-monitoring/issue-owners/), a team, or a member.
 - Send a notification to an integration, which can include these options, depending on which integrations you have installed:
-  - Send a [Slack](/product/integrations/slack/) notification
-  - Send a [PagerDuty](/product/integrations/pagerduty/) notification
-  - Send a [Microsoft Teams](/product/integrations/msteams/) notification
+  - Send a [Slack](/product/integrations/notification-incidents/slack/) notification
+  - Send a [PagerDuty](/product//integrations/notification-incidents/pagerduty/) notification
```
Comment (imatwawana) [inline src/docs/product/alerts/create-alerts/issue-alert-config.mdx:62]:
```suggestion
  - Send a [PagerDuty](/product/integrations/notification-incidents/pagerduty/) notification
```

File: src/docs/product/integrations/index.mdx [hunk 2]
```diff
@@ -23,13 +23,14 @@ For more details, see the [full Integration Platform documentation](/product/int
...
 | Datadog                                           | X            |               |                |
+| [Microsoft Teams](/product/integrations/notification-incidents/msteams/) | X            | X             |                |
 | OpsGenie                                          | X            |               |                |
-| [PagerDuty](/product/integrations/pagerduty/)     | X            | X             |                |
+| [PagerDuty](/product/integrations/notification-incidents/pagerduty/)     | X            | X             |                |
 | Pushover                                          | X            |               |                |
-| [Slack](/product/integrations/slack/)             | X            | X             | X              |
-| [Microsoft Teams](/product/integrations/msteams/) | X            | X             |                |
+| [Slack](/product/integrations/notification-incidents/slack/)             | X            | X             | X              |
+| [Spike.sh](/product/integrations/notification-incidents/slack/)          |              |               |                |
```
Comment (imatwawana) [inline src/docs/product/integrations/index.mdx:33]: @manuzope - I'm adding Spike.sh to the table on the main Integrations page. Is this the right table for it to be in on that page (Notification & Incidents) and if so, which features does it have in terms of the table columns?

**Review by PeloWriter — APPROVED** [review]
So much better. Thank you!

File: src/docs/product/integrations/index.mdx (hunk 2 above)
Comment (manuzope) [reply src/docs/product/integrations/index.mdx:33]: Ah yes, thank you for doing this. This is the correct table. It has issue alerts and metric alerts, not link unfurling.

Comment (imatwawana) [reply src/docs/product/integrations/index.mdx:33]: Thank you!!

Comment (imatwawana) [inline src/docs/product/integrations/index.mdx:33]:
```suggestion
| [Spike.sh](/product/integrations/notification-incidents/slack/)          | X             | X              |                |
```

**Pull request merged**